src/components/Radio/Radio.tsx

```diff
@@ -48,11 +48,7 @@ class Radio extends UIComponent<any, any> {
     return (
       <ElementType {...rest} className={classes.root}>
         <Label styles={{ root: styles.label }}>
-          {createHTMLInput(type, {
-            overrideProps: {
-              className: classes.radio,
-            },
-          })}
+          <Input type="radio" styles={{ root: styles.radio }} />
```

Comment:
I'm wary of this composition here.  I think before we start to compose these two components we should first get the Radio's styles and accessibility working.  My fear is that the logic, layout, and styles required for the radio will not make this kind of composition less desirable than using a vanilla `input` here.

Reply:
this puts a very important point about our vision for the components that use the same name and, essentially, semantics as their HTML counterparts. 

My thought on that is that it seems to be reasonable to provide client with the same behavior and visual aspects as those are for HTML element in case if the same set of attributes is used, i.e.

```jsx
<Input type='radio' />

// should be essentially equivalent to use of
<input type='radio'>
```

At the same time, those components are designed to introduce augmented functionaliy - so the set of the things that can do is a superset of their HTML analogs. For example, `Input` component is able to properly handle cases where `label` prop is provided:

```jsx
<Input type='radio'  label='foo value' />    // renders proper HTML
```
Otherwise there would be too many surprises provided to client if she will start to consume our components. Suppose that there is an application where forms were used - and at some point its developer decides to integrate Stardust. Why? For example, to extend `input`s with our clearable fuinctionality. What could be the first step? Replace `input` elements on `Input` components. But if this change would produce some drastic changes to the look and presentational aspects of the application, this will be a blocker for further progress.

And, if we agree on these points mentioned above, it seems to be reasonable approach to guarantee this semantics consistency by consuming our components inside other ones, so that any problems will be immediately visible.

--------

Hope that I am rather clear with reasoning behind my thoughts on that. Please, share your opinion in return